Library/Homebrew/cmd/update.sh

```diff
@@ -336,23 +349,41 @@ EOS
         git fetch --force "${QUIET_ARGS[@]}" origin \
           "refs/heads/$UPSTREAM_BRANCH:refs/remotes/origin/$UPSTREAM_BRANCH" 2>/dev/null
       else
-        git fetch --force "${QUIET_ARGS[@]}" origin \
-          "refs/heads/$UPSTREAM_BRANCH:refs/remotes/origin/$UPSTREAM_BRANCH" || \
-            odie "Fetching $DIR failed!"
+        if ! git fetch --force "${QUIET_ARGS[@]}" origin \
+          "refs/heads/$UPSTREAM_BRANCH:refs/remotes/origin/$UPSTREAM_BRANCH"
+        then
+          echo "Fetching $DIR failed!" >> "$HOMEBREW_UPDATE_FAILED_FILE"
+        fi
       fi
     ) &
   done
 
   wait
   trap - SIGINT
 
+  if [[ -f "$HOMEBREW_UPDATE_FAILED_FILE" ]]
+  then
+    onoe < "$HOMEBREW_UPDATE_FAILED_FILE"
+    rm -f "$HOMEBREW_UPDATE_FAILED_FILE"
+    export HOMEBREW_UPDATE_FAILED="1"
+  fi
+
+  unset HOMEBREW_UPDATED
```

Comment:
I wonder if this should have a comment briefly explaining how this is used, as the variable is evaluated just a few lines below after apparently being unset. It might not be immediately clear that it is modified in the `pull` function.